File: java/src/test/java/problems201_300/FirstBadVersionTest.java
package problems201_300;

import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertThat;

public class FirstBadVersionTest extends SampledTest {
    @Test
    public void test() {
        for (FirstBadVersionSample sample
            : getSamples(FirstBadVersionSample[].class)) {
            assertThat(new FirstBadVersion(sample.input.bad).firstBadVersion(sample.input.n))
                .isEqualTo(sample.output);
        }
    }
}

File: kotlin/src/test/kotlin/problems201_300/FirstBadVersionTest.kt
package problems201_300

import com.google.common.truth.Truth.assertThat
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class FirstBadVersionTest : SampledTest() {
    @Test
    fun test() =
        getSamples<Array<FirstBadVersionSample>>().forEach { (input, output, _) ->
            assertThat(FirstBadVersion(input.bad).firstBadVersion(input.n))
                .isEqualTo(output)
        }
}
